components/script/dom/performance.rs

```diff
@@ -51,6 +52,16 @@ impl PerformanceEntryList {
             entry_type.as_ref().map_or(true, |type_| *e.entry_type() == *type_)
         ).map(|e| e.clone()).collect()
     }
+
+    pub fn get_entries_by_name_and_type(&self, name: Option<DOMString>, entry_type: Option<DOMString>)
```

Comment:
Thank you for implementing this algorithm! I filed #18186 to update the code so all the getters use this implementation.